### PR TITLE
Swagger doc

### PR DESCRIPTION
**Pull request by asimpat**

### CODE REVIEW

File: src/api/modules/auth/Guards/roleguard.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+export class RolesGuard implements CanActivate {
+  constructor(private reflector: Reflector) {}
+
+  canActivate(context: ExecutionContext): boolean {
+    const roles = this.reflector.get<string[]>('roles', context.getHandler());
+    if (!roles) {
+      return true;
+    }
+    const request = context.switchToHttp().getRequest();
+      const user = request.user;
```
Comment (elishaukpong) [inline src/api/modules/auth/Guards/roleguard.ts:14]: variable alignments should be consistent. @asimpat

File: src/api/modules/auth/Guards/roleguard.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      
+      const hasrole = roles.indexOf(user.role) > -1 
+      if (hasrole) {
```
Comment (elishaukpong) [inline src/api/modules/auth/Guards/roleguard.ts:17]: there is no use for the hasRole variable, just return the logic directly

`return roles.indexOf(user.role) > -1;`


It is important that you use `;` at the end of your lines, learn it incase you go to other languages that are strict with it, you wont struggle with it

File: src/api/modules/auth/Guards/userGuard.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { Injectable, CanActivate, ExecutionContext } from '@nestjs/common';
+import { Observable } from 'rxjs';
+
+@Injectable()
+export class UserGuard implements CanActivate {
+  canActivate(
+    context: ExecutionContext,
```
Comment (elishaukpong) [inline src/api/modules/auth/Guards/userGuard.ts:7]: no need bringing this to the next line

File: src/api/modules/auth/Guards/userGuard.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  ): boolean | Promise<boolean> | Observable<boolean> {
+      const request = context.switchToHttp().getRequest();
+      const user = request.user
```
Comment (elishaukpong) [inline src/api/modules/auth/Guards/userGuard.ts:10]: this variable is redundant

File: src/api/modules/auth/Guards/userGuard.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+      if (user.role == 'user') {
```
Comment (elishaukpong) [inline src/api/modules/auth/Guards/userGuard.ts:12]: use `return user.role == 'user';`

no need for the `if` statement

File: src/api/modules/auth/Guards/userGuard.ts (hunk 4 above)
Comment (JC-Coder) [reply src/api/modules/auth/Guards/userGuard.ts:10]: sir please how do you mean. I don't understand

Comment (elishaukpong) [reply src/api/modules/auth/Guards/userGuard.ts:10]: the variable user is not really needed there

you can use `request.user` directly.

in your code, if you are not using a variable In more than one place, then don't make a variable, use that value directly

**Comment by asimpat** [comment]
Sent from Yahoo Mail on Android 
 
  On Tue, 11 Oct 2022 at 9:58 pm, Elisha ***@***.***> wrote:   


@elishaukpong commented on this pull request.

In src/api/modules/auth/Guards/roleguard.ts:
> @@ -0,0 +1,23 @@
+import { Injectable, CanActivate, ExecutionContext } from ***@***.***/common';
+import { Reflector } from ***@***.***/core';
+
***@***.***()
+export class RolesGuard implements CanActivate {
+  constructor(private reflector: Reflector) {}
+
+  canActivate(context: ExecutionContext): boolean {
+    const roles = this.reflector.get<string[]>('roles', context.getHandler());
+    if (!roles) {
+      return true;
+    }
+    const request = context.switchToHttp().getRequest();
+      const user = request.user;

variable alignments should be consistent. @asimpat

In src/api/modules/auth/Guards/roleguard.ts:
> +import { Reflector } from ***@***.***/core';
+
***@***.***()
+export class RolesGuard implements CanActivate {
+  constructor(private reflector: Reflector) {}
+
+  canActivate(context: ExecutionContext): boolean {
+    const roles = this.reflector.get<string[]>('roles', context.getHandler());
+    if (!roles) {
+      return true;
+    }
+    const request = context.switchToHttp().getRequest();
+      const user = request.user;
+      
+      const hasrole = roles.indexOf(user.role) > -1 
+      if (hasrole) { 

there is no use for the hasRole variable, just return the logic directly

return roles.indexOf(user.role) > -1;

It is important that you use ; at the end of your lines, learn it incase you go to other languages that are strict with it, you wont struggle with it

In src/api/modules/auth/Guards/userGuard.ts:
> @@ -0,0 +1,18 @@
+import { Injectable, CanActivate, ExecutionContext } from ***@***.***/common';
+import { Observable } from 'rxjs';
+
***@***.***()
+export class UserGuard implements CanActivate {
+  canActivate(
+    context: ExecutionContext,

no need bringing this to the next line

In src/api/modules/auth/Guards/userGuard.ts:
> @@ -0,0 +1,18 @@
+import { Injectable, CanActivate, ExecutionContext } from ***@***.***/common';
+import { Observable } from 'rxjs';
+
***@***.***()
+export class UserGuard implements CanActivate {
+  canActivate(
+    context: ExecutionContext,
+  ): boolean | Promise<boolean> | Observable<boolean> {
+      const request = context.switchToHttp().getRequest();
+      const user = request.user

this variable is redundant

If we remove the variable, How about the *user.role = "user"* sir







In src/api/modules/auth/Guards/userGuard.ts:
> @@ -0,0 +1,18 @@
+import { Injectable, CanActivate, ExecutionContext } from ***@***.***/common';
+import { Observable } from 'rxjs';
+
***@***.***()
+export class UserGuard implements CanActivate {
+  canActivate(
+    context: ExecutionContext,
+  ): boolean | Promise<boolean> | Observable<boolean> {
+      const request = context.switchToHttp().getRequest();
+      const user = request.user
+
+      if (user.role == 'user') {

use return user.role == 'user';

no need for the if statement

**Pull request merged**